lib/animina/accounts/resources/basic_user.ex

```diff
@@ -42,7 +42,14 @@ defmodule Animina.Accounts.BasicUser do
                   allow_empty?: false
     end
 
-    attribute :gender, :string, allow_nil?: false
+    attribute :gender, :string do
+      allow_nil? false
+      
+      constraints(
+        match: ~r/\A [mfx] \z/x,
```

Comment:
In case there are no technical reasons I would prefer `male`, `female` and `diverse` over `m`, `f` and `x`. Less thinking is needed while reading the code by using the full words.

Reply:
If this is about personal preference, then I don't want to argue about it.

BTW: If I'm not mistaken the actual data value was "`Diverse / non-binary`" before this change.

OTOH if this is about technical reasons, then I don't see a point of using way more space in the database table and index than necessary.

I would suggest to mentally separate data from how you work with the objects. I.e. one could probably add something like this to the user model (pseudo-code):

```ex
def male?    ; gender == "m"; end
def female?  ; gender == "f"; end
def diverse? ; gender == "x"; end
```

But I think that won't even be necessary, since probably the only other place where the gender matters will be the matching algorithm, and there will likely not even be any literal "`male`"/"`female`"/"`diverse_or_non_binary`" in the code, as the algorithm will simply match data.

Actually the database column should be an `ENUM` instead of a `TEXT` field, but that's outside the scope of this issue. Also, there should be an index on the database column, or rather a combined index on e.g geolocation and gender or something, depending on the queries that the matching algorithm will run.

Reply:
Not tested code:

```elixir
defmodule YourApp.Repo.Migrations.AddGenderType do
  use Ecto.Migration

  def up do
    execute "CREATE TYPE gender AS ENUM ('male', 'female', 'diverse')"
  end

  def down do
    execute "DROP TYPE gender"
  end
end
```

```elixir
defmodule YourApp.Accounts.User do
  use Ecto.Schema
  import Ecto.Changeset

  schema "users" do
    field :name, :string
    field :gender, :string, comment: "ENUM('male', 'female', 'diverse')"
    # Andere Felder...
  end

  def changeset(user, attrs) do
    user
    |> cast(attrs, [:name, :gender])
    |> validate_inclusion(:gender, ["male", "female", "diverse"])
  end
end
```

@philipp-kempgen Fire at will (which includes changing nothing at all). We do have bigger fish to fry but if you want to tackle this: Go ahead.